src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,600 @@
+  user-select: none;
+  overflow-anchor: none;
+`;
+
+const getRenderedCursor = (children: Array<ReactElement>) =>
+  children.reduce(
+    ([minRow, maxRow, minColumn, maxColumn], { props: { columnIndex, rowIndex } }) => {
+      if (rowIndex < minRow) {
+        minRow = rowIndex;
+      }
+      if (rowIndex > maxRow) {
+        maxRow = rowIndex;
+      }
+      if (columnIndex < minColumn) {
+        minColumn = columnIndex;
+      }
+      if (columnIndex > maxColumn) {
+        maxColumn = columnIndex;
+      }
+
+      return [minRow, maxRow, minColumn, maxColumn];
+    },
+    [
+      Number.POSITIVE_INFINITY,
+      Number.NEGATIVE_INFINITY,
+      Number.POSITIVE_INFINITY,
+      Number.NEGATIVE_INFINITY,
+    ]
+  );
+
+const GridDataContainer = styled.div<{ $top: number; $left: number }>`
+  position: absolute;
+  top: 0;
+  left: 0;
+  ${({ $top, $left }) => `
+    margin-top: ${$top}px;
+    margin-left: ${$left}px;
+  `}
+`;
+
+const ContextMenuTrigger = styled.div`
+  outline: none;
+  height: 100%;
+  width: 100%;
+`;
+
+interface InnerElementTypeTypes extends HTMLAttributes<HTMLDivElement> {
+  children: Array<ReactElement>;
+}
+
+const OuterElementType = forwardRef<HTMLDivElement>((props, ref) => (
+  <div
+    ref={ref}
+    data-testid="grid-outer-element"
+    {...props}
+  />
+));
+
+export const Grid = forwardRef<VariableSizeGrid, GridProps>(
+  (
+    {
+      rowStart = 0,
+      showRowNumber = true,
+      rounded = "none",
+      showHeader = true,
+      focus,
+      useIsScrolling = true,
+      rowHeight = 33,
+      columnCount,
+      columnWidth,
+      onSelect: onSelectProp,
+      headerHeight = 33,
+      rowCount,
+      cell,
+      onColumnResize: onColumnResizeProp,
+      onFocusChange: onFocusChangeProp,
+      getMenuOptions,
+      onKeyDown: onKeyDownProp,
+      ...props
+    },
+    forwardedRed
+  ) => {
+    const [menuOptions, setMenuOptions] = useState<Array<GridContextMenuItemProps>>([]);
+    const [contextMenuOpen, setContextMenuOpen] = useState(false);
+    const [scrolledVertical, setScrolledVertical] = useState(false);
+    const [scrolledHorizontal, setScrolledHorizontal] = useState(false);
+    const dragState = useRef<number | false>(false);
+    const gridRef = useRef<VariableSizeGrid>(null);
+    const containerRef = useRef<HTMLDivElement>(null);
+    const outerRef = useRef<HTMLDivElement>(null);
+    const elementBorderRef = useRef({
+      top: 0,
+      bottom: 0,
+      left: 0,
+      right: 0,
+      scrollBarWidth: 0,
+      scrollBarHeight: 0,
+      width: 0,
+      height: 0,
+    });
+    const onCellSelect: onSelectFn = useCallback(
+      (action, selection, focus) => {
+        if (typeof onSelectProp === "function") {
+          onSelectProp(action, selection, focus);
+        }
+        if (typeof getMenuOptions === "function") {
+          setMenuOptions(getMenuOptions(selection, focus));
```

Comment:
instead of setting the menu options every time a new cell is selected, the component should have a selection state. When the context menu options is triggered it should read that selection state